src/models/notifications.js

```diff
@@ -0,0 +1,43 @@
+'use strict'
+
+const log = require('../services/log')('subscriptions')
+const uri = require('../services/uriManager')
+const NotFoundError = require('five-bells-shared/errors/not-found-error')
+const UnauthorizedError = require('five-bells-shared/errors/unauthorized-error')
+const Notification = require('./db/notification').Notification
+const Transfer = require('./db/transfer').Transfer
+
+function isOwnerOrAdmin (requestingUser, subscription) {
+  const requestOwner = uri.make('account', requestingUser.name)
+  return requestOwner === subscription.owner || requestingUser.is_admin
+}
+
+function * getNotification (subscriptionId, notificationId, requestingUser) {
+  log.debug('fetching notification ID ' + notificationId)
+  const notification = yield Notification.findOne({
+    where: {
+      id: notificationId
+    },
+    include: [{ all: true }]})
+  if (!notification) {
+    throw new NotFoundError('Unknown notification ID')
+  } else if (notification.Subscription.id !== subscriptionId) {
```

Comment:
This will throw if `notification.Subscription` is `undefined`, which should never happen, but is possible if the database becomes corrupted. I can add a check for this case, if we think it's worthwhile.